src/main/scala/com/twitter/finagle/WebSocket.scala

```diff
@@ -1,13 +1,45 @@
 package com.twitter.finagle
 
+import com.twitter.finagle.netty3._
 import com.twitter.finagle.param.{Label, ProtocolLibrary, Stats}
+import com.twitter.finagle.client.{Transporter, StdStackClient, StackClient}
 import com.twitter.finagle.server.{Listener, StdStackServer, StackServer}
 import com.twitter.finagle.transport.Transport
-import com.twitter.finagle.websocket.{Netty3, Request, Response, ServerDispatcher}
+import com.twitter.finagle.websocket.{Netty3, ClientDispatcher, Request, Response, ServerDispatcher}
 import com.twitter.util.Closable
 import java.net.SocketAddress
+import org.jboss.netty.channel.Channel
 
 object Websocket extends Server[Request, Response] {
+  case class Client(
+    stack: Stack[ServiceFactory[Request, Response]] = StackClient.newStack,
```

Comment:
TODO: Incorporate parts of the HTTP stack relevant to streaming (e.g. DelayedServiceFactory).